src/DocTest/Parser.elm

```diff
@@ -48,6 +53,11 @@ commentRegex =
     Regex.regex "{-[^]*?-}"
 
 
+innerCommentRegex : Regex
+innerCommentRegex =
+    Regex.regex "--(.*)"
```

Comment:
A safer solution would be to change how we compile the actual tests.

If you change `toTest` in `DocTest.Compiler` to:

```elm

toTest : Test -> String
toTest test =
    String.join "\n"
        [ indent 2 "Test.test \">>> " ++ escape test.assertion ++ "\" <|"
        , indent 3 "\\() ->"
        , indent 4 "("
        , indent 5 test.assertion
        , indent 4 ") |> Expect.equal ("
        , indent 5 test.expectation
        , indent 4 ")"
        ]
```
Everything works.
The created tests look like this:

```elm
        Test.test ">>> quux 10 32 -- another" <|
            \() ->
                (
                    quux 10 32 -- another
                )
                |> Expect.equal (
                    42 --result
                ),
        Test.test ">>> silly \"--\" \"foo\" -- another" <|
            \() ->
                (
                    silly "--" "foo" -- another
                )
                |> Expect.equal (
                    "foo--"
                )

```